File: Animator/C#/PatternsAnimator/Lib/Mover.cs
using PatternsAnimator.Utils;

namespace PatternsAnimator.Lib
{
    public class Mover : EffectBase<IMovable>
    {
        private readonly Point _p;

        public Mover(Point p)
        {
            _p = p;
        }

        protected override void DoAnimation(IMovable o)
        {
            o.SetPosition(_p);
        }
    }
}

File: Animator/Java/src/Lib/Mover.java
package Lib;

import Utils.Point;

public class Mover extends EffectBase<IMovable> {
    private final Point position;

    public Mover(Point position) {
        this.position = position;
    }

    @Override
    protected void doAnimation(IMovable o) {
        o.setPosition(this.position);
    }
}
